src/ring/middleware/authorization.clj

```diff
@@ -0,0 +1,84 @@
+(ns ring.middleware.authorization
+  (:refer-clojure :exclude [update-keys])
+  (:require
+    [clojure.string :as str]
+    [ring.util.parsing :as parsing]))
+
+;; Copied from Clojure 1.11:
+(defn- update-keys
+  [m f]
+  (let [ret (persistent!
+              (reduce-kv (fn [acc k v] (assoc! acc (f k) v))
+                         (transient {})
+                         m))]
+    (with-meta ret (meta m))))
+
+(def ^{:added "0.4"} re-token68
```

Comment:
I don't think we want this to be part of the API. We should make it private for now and cut the docstring. Also the `:added` metadata is generally used for post-1.0, so I think it can be removed too.